api/auth_token.go

```diff
@@ -179,6 +215,13 @@ func (c *TokenAuth) RenewSelf(increment int) (*Secret, error) {
 // RenewTokenAsSelf behaves like renew-self, but authenticates using a provided
 // token instead of the token attached to the client.
 func (c *TokenAuth) RenewTokenAsSelf(token string, increment int) (*Secret, error) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	return c.RenewTokenAsSelfWithContext(ctx, token, increment)
+}
+
+// RenewTokenAsSelfWithContext the same as RenewTokenAsSelf, but with a custom context.
```

Comment:
This is a bit of a nit but would it be possible to move the documentation for each pair of methods into the `__WithContext(...)` one, since it will be the primary way of calling the API moving forward.

The wrapper function could have a simple one-liner like the following:
```
// RenewTokenAsSelf wraps RenewTokenAsSelfWithContext using context.Background.
```

This is consistent with the [net/http NewRequest + NewRequestWithContext](https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/net/http/request.go;l=829) documentation.

Reply:
Done

Reply:
Thanks for making the change! It looks like a few extra `// //` sneaked into the comments:

https://github.com/hashicorp/vault/blob/af651acc743aa378bfe53397797a5b38c26f77c6/api/auth_token.go#L223

Reply:
fixed